src/pages/List.js

```diff
@@ -5,11 +5,34 @@ import useListToken from '../useListToken';
 import { FirestoreCollection } from 'react-firestore';
 import Loading from '../components/Loading';
 import ErrorMessage from '../components/ErrorMessage';
+import HomePageButton from '../components/HomePageButton';
+import dayjs from 'dayjs';
+import './List.css';
+
+const today = dayjs();
+
+function isLessThan24hrs(datePurchased) {
+  let purchaseDateCalc = dayjs(datePurchased);
+  return today.diff(purchaseDateCalc, 'hour') <= 24;
+}
 
 const List = props => {
-  const { shoppingList, setShoppingList } = useContext(ListContext);
+  const { shoppingList, setShoppingList, addDatePurchased } = useContext(
+    ListContext,
+  );
   const { token } = useListToken();
 
+  //we are checking if the last date it was purchased is less than 24hrs using isLessThan24hrs function
+  function isChecked(lastDatePurchased) {
+    return !!lastDatePurchased && isLessThan24hrs(lastDatePurchased);
```

Comment:
`!!lastDatePurchased` because `lastDatePurchased` starts out as `undefined` (or `null`?) when an item has been created but not yet purchased. Is that correct?

Might be worth leaving a comment to that effect!